### PR TITLE
Wrap derivation_path::DerivationPath

### PR DESCRIPTION
**Pull request by CriesofCarrots**
#### Problem
Sdk/remote-wallet uses a hand-rolled DerivationPath, which makes future integration with ed25519-dalek-bip32 difficult.

#### Summary of Changes
Replace DerivationPath with one that wraps `derivation_path::DerivationPath`
Add helpers for solana bip44 derivation, and derivations from query value strings

Toward #5246

### CODE REVIEW

File: sdk/src/derivation_path.rs [hunk 1]
```diff
@@ -99,65 +108,158 @@ impl DerivationPath {
...
+}
+
+fn bip44_solana_indexes() -> Vec<ChildIndex> {
+    vec![
+        ChildIndex::Hardened(BIP_44_PURPOSE),
+        ChildIndex::Hardened(BIP_44_COIN_SOL),
+    ]
+}
+
+fn child_index_from_str(s: &str) -> Result<ChildIndex, DerivationPathError> {
```
Comment (t-nelson) [inline sdk/src/derivation_path.rs]: If you like the above suggestions, I don't think this has any more callers

**Review by t-nelson — COMMENTED** [review]
Looking good!  I added some suggestions for foisting more work onto `derivation-path`

File: sdk/src/derivation_path.rs [hunk 2]
```diff
@@ -99,65 +108,158 @@ impl DerivationPath {
...
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "m")?;
+        for index in self.0.path() {
+            write!(f, "/{}", index)?;
+        }
+        Ok(())
+    }
+}
+
+fn bip44_solana_indexes() -> Vec<ChildIndex> {
```
[the remark this answers is not in the document]
Comment (CriesofCarrots) [reply sdk/src/derivation_path.rs]: I just pushed a commit that replaces this with a `Bip44` trait. Wdyt? Hoping to make it more extensible.

File: sdk/src/derivation_path.rs [hunk 3]
```diff
@@ -49,6 +49,30 @@ impl DerivationPath {
...
+                    s.push_str("\'");
+                }
+                s
+            })
+            .collect::<Vec<_>>()
+            .join("/");
+        Ok(Self(DerivationPathInner::from_str(&string).map_err(
+            |err| DerivationPathError::InvalidDerivationPath(err.to_string()),
+        )?))
+    }
```
[the remark this answers is not in the document]
Comment (CriesofCarrots) [reply sdk/src/derivation_path.rs:68]: Oh, that seems nice. Let me take it for a test drive!

File: sdk/src/derivation_path.rs (hunk 1 above)
Comment (CriesofCarrots) [reply sdk/src/derivation_path.rs]: Indeed! \o/

File: sdk/src/derivation_path.rs [hunk 4]
```diff
@@ -118,67 +119,82 @@ impl fmt::Debug for DerivationPath {
...
+
+    fn base_indexes(&self) -> Vec<ChildIndex> {
+        vec![
+            ChildIndex::Hardened(Self::PURPOSE),
+            ChildIndex::Hardened(Self::COIN),
+        ]
+    }
+}
+
+pub struct Solana;
```
Comment (t-nelson) [inline sdk/src/derivation_path.rs:128]: Do you think we're going to end up having to maintain a bunch of these?

**Review by t-nelson — COMMENTED** [review]
I like the structural changes of Bip44 as a trait.  A little concerned about the potential maintenance burden of the implementers though :thinking:

File: sdk/src/derivation_path.rs (hunk 4 above)
Comment (CriesofCarrots) [reply sdk/src/derivation_path.rs:128]: I hope not. But maybe we can set up a framework for `DerivationPath` callers (eg `clap-utils`) that pulls from a registry of `impl Bip44` definitions...?

Comment (CriesofCarrots) [reply sdk/src/derivation_path.rs:128]: I wouldn't put any other than this one in sdk, for sure

File: sdk/src/derivation_path.rs [hunk 5]
```diff
@@ -1,94 +1,90 @@
...
+        Ok(Self(DerivationPathInner::from_str(&path).map_err(
+            |err| DerivationPathError::InvalidDerivationPath(err.to_string()),
+        )?))
     }
-}
 
-impl std::fmt::Debug for DerivationPathComponent {
-    fn fmt(&self, fmt: &mut std::fmt::Formatter) -> std::fmt::Result {
-        std::fmt::Display::fmt(self, fmt)
+    pub fn new_bip44<T: Bip44>(coin: T, account: Option<u32>, change: Option<u32>) -> Self {
```
Comment (t-nelson) [inline sdk/src/derivation_path.rs:62]: Let's rename this to `new_bip44_with_coin`, add a pub wrapper that calls with `Solana` and make all of the trait referencing bits private for now.  I'm tempted to say our API referencing BIP44 can assume the Solana address and any other implementer can take the `full-path`/BIP32 route.  But I'm no so confident as to reject the trait at this point

**Review by t-nelson — APPROVED** [review]
Let's make the trait bits private for now. LGTM with that change!

File: sdk/src/derivation_path.rs [hunk 6]
```diff
@@ -99,65 +103,166 @@ impl DerivationPath {
...
+            ChildIndex::Hardened(Self::COIN),
+        ]
+    }
+}
+
+struct Solana;
+
+impl Bip44 for Solana {
+    const COIN: u32 = 501;
+}
```
Comment (roman-tik) [inline sdk/src/derivation_path.rs:140]: let's do the same, but for Velas derivation path

**Pull request merged**